snooty/postprocess.py

```diff
@@ -104,6 +105,30 @@ def run(
         # Update metadata document with key-value pairs defined in event parser
         document.update({"slugToTitle": self.slug_title_mapping})
 
+        def clean_and_validate_page_group_slug(slug: str) -> Optional[str]:
+            """Clean a slug and validate that it is a known page. If it is not, return None."""
+            cleaned = clean_slug(slug)
+            if cleaned not in self.slug_title_mapping:
+                # XXX: Because reporting errors in config.toml properly is dodgy right now, just
+                # log to stderr.
+                logger.error(f"Cannot find slug '{cleaned}'")
+                return None
+
+            return cleaned
+
+        # Normalize all page group slugs
+        page_groups = {
+            title: [
+                slug
+                for slug in (clean_and_validate_page_group_slug(slug) for slug in slugs)
+                if slug
+            ]
+            for title, slugs in self.project_config.page_groups.items()
+        }
+
+        if page_groups:
+            document.update({"pageGroups": page_groups})
+
```

Comment:
Because I don't know of any plans to use this type of content in Docs, it may make sense instead include this in `DevhubPostprocessor.run` from #96 and avoid including it in a normal docs parse. Thoughts?

Reply:
Makes sense! I'll rebase as soon as #96 is merged!